internal/printer/conditions.go

```diff
@@ -0,0 +1,128 @@
+/*
+Copyright (c) 2021 the Octant contributors. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package printer
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/vmware-tanzu/octant/pkg/view/component"
+	"github.com/vmware-tanzu/octant/pkg/view/flexlayout"
+)
+
+var (
+	conditionType           = "Type"
+	conditionReason         = "Reason"
+	conditionStatus         = "Status"
+	conditionMessage        = "Message"
+	conditionLastUpdate     = "Last Update"
+	conditionLastTransition = "Last Transition"
+
+	conditionMap = [][]string{
+		{conditionType, "type"},
+		{conditionReason, "reason"},
+		{conditionStatus, "status"},
+		{conditionMessage, "message"},
+		{conditionLastUpdate, "lastUpdateTime"},
+		{conditionLastTransition, "lastTransitionTime"},
+	}
+)
+
+// createConditionsTable returns a component.Table, if conditions exist on the object, and any err.
+// For objects with empty conditions, an empty table with placeholder text is used.
+func createConditionsTable(u unstructured.Unstructured, sortKey string, customConditionMap [][]string) (*component.Table, bool, error) {
```

Comment:
I think this shouldn't return a bool and an exportable `HasConditions(u unstructured.Unstructured)` should be available to check rather than always attempting to make a table

Reply:
I agree with essence of this idea, I'll push up a change that I think satisfies this.